contracts/transfer/src/wasm/internal.rs

```diff
@@ -4,74 +4,63 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-use crate::{PublicKeyBytes, TransferContract};
-use core::convert::TryFrom;
+use crate::{Error, Map, PublicKeyBytes, TransferContract};
 
 use alloc::vec::Vec;
-use canonical::{InvalidEncoding, Store};
+use dusk_abi::ContractId;
 use dusk_bls12_381::BlsScalar;
 use dusk_bytes::Serializable;
 use dusk_jubjub::JubJubAffine;
-use dusk_kelvin_map::Map;
-use dusk_pki::PublicKey;
+use dusk_pki::{PublicKey, PublicSpendKey};
 use phoenix_core::{Crossover, Fee, Message, Note};
-use rusk_abi::PublicInput;
+use rusk_abi::{PaymentInfo, PublicInput};
 
-impl<S: Store> TransferContract<S> {
-    pub(crate) fn push_fee_crossover(
-        &mut self,
-        fee: Fee,
-    ) -> Result<(), S::Error> {
+impl TransferContract {
+    pub(crate) fn push_fee_crossover(&mut self, fee: Fee) -> Result<(), Error> {
         let block_height = dusk_abi::block_height();
 
         // FIXME Get gas consumed
         // https://github.com/dusk-network/rusk/issues/195
         let gas_consumed = 2;
         let remainder = fee.gen_remainder(gas_consumed);
         let remainder = Note::from(remainder);
-        let remainder_value =
-            remainder.value(None).map_err(|_| InvalidEncoding.into())?;
+        let remainder_value = remainder.value(None)?;
         if remainder_value > 0 {
             self.push_note(block_height, remainder)?;
         }
 
         if let Some(crossover) = self.var_crossover {
-            Note::try_from((fee, crossover))
-                .map_err(|_| InvalidEncoding.into())
-                .and_then(|note| self.push_note(block_height, note))?;
+            let note = Note::from((fee, crossover));
+            self.push_note(block_height, note)?;
         }
 
         Ok(())
     }
 
-    // TODO convert to const fn
-    // https://github.com/rust-lang/rust/issues/57563
-    pub(crate) fn minimum_gas_price() -> u64 {
-        // FIXME define the mininum gas price
-        // https://github.com/dusk-network/rusk/issues/195
-        0
+    pub(crate) const fn minimum_gas_price() -> u64 {
+        // TODO link the docs
+        1
```

Comment:
This is not contract's responsability but VM's one. We should save this info there since on the contract we can never change it if we want to in the future.

Reply:
AFAIK, we don't have such function available in the ABI

Reply:
As discussed via call, its applicable and we just need to append documents and mention this is always in nano-dusk

Reply:
Fixed